service/src/test/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperServiceTest.java

```diff
@@ -834,6 +895,316 @@ void shouldAddCustomKeyValuePairWhenDirectionsBelongToCourt() {
+        }
+
+        // LOCAL_AUTHORITY would be added here if the functionality for complying on behalf of LA is implemented.
+        @ParameterizedTest
+        @EnumSource(value = DirectionAssignee.class, names = {"CAFCASS"})
+        void shouldPopulateDirectionsWhenSingleResponseDirections(DirectionAssignee assignee) {
+            CaseDetails caseDetails = CaseDetails.builder().data(new HashMap<>()).build();
+            Map<DirectionAssignee, List<Element<Direction>>> directionsMap = new HashMap<>();
+            directionsMap.put(assignee, buildDirections(assignee));
+            directionsMap.put(ALL_PARTIES, buildDirections(ALL_PARTIES));
+
+            service.addDirectionsToCaseDetails(caseDetails, directionsMap);
+
+            List<Element<Direction>> expectedDirections = buildDirections(assignee);
+            expectedDirections.addAll(buildDirections(ALL_PARTIES));
+
+            assertThat(caseDetails).isEqualTo(CaseDetails.builder()
+                .data(ImmutableMap.of(assignee.getValue().concat("Custom"), expectedDirections))
+                .build());
+        }
+    }
+
+    @Nested
+    class AddComplyOnBehalfResponsesToDirectionsInStandardDirectionOrder {
+
+        @Test
+        void shouldAddCafcassResponseWhenValidResponseMadeByCourt() {
+            UUID directionId = randomUUID();
+
+            Order sdo = Order.builder()
+                .directions(ImmutableList.of(Element.<Direction>builder()
+                    .id(directionId)
+                    .value(Direction.builder()
+                        .directionType("example direction")
+                        .assignee(CAFCASS)
+                        .build())
+                    .build()))
+                .build();
+
+            List<Element<Direction>> directionWithResponse = ImmutableList.of(Element.<Direction>builder()
+                .id(directionId)
+                .value(Direction.builder()
+                    .response(DirectionResponse.builder()
+                        .directionId(directionId)
+                        .assignee(CAFCASS)
+                        .complied("Yes")
+                        .build())
+                    .build())
+                .build());
+
+            CaseData caseData = CaseData.builder()
+                .standardDirectionOrder(sdo)
+                .cafcassDirectionsCustom(directionWithResponse)
+                .build();
+
+            DirectionResponse expectedResponse = DirectionResponse.builder()
+                .directionId(directionId)
+                .assignee(COURT)
+                .respondingOnBehalfOf("CAFCASS")
+                .complied("Yes")
+                .build();
+
+            service.addComplyOnBehalfResponsesToDirectionsInStandardDirectionsOrder(caseData);
+
+            assertThat(
+                caseData.getStandardDirectionOrder().getDirections().get(0).getValue().getResponses().get(0).getValue())
+                .isEqualTo(expectedResponse);
+        }
+
+        @Test
```

Comment:
1237 lines of tests, smells like class doing too many things.

Reply:
I agree, the directionHelperService has become enormous. Outside the scope of this story but I think I would like to split into services: a service which sorts/filters directions, a service which adds directions to case details, and a service that persists data for hidden fields in ccd.